sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -46,6 +46,7 @@
 class ServiceBusReceiverAsyncClientIntegrationTest extends IntegrationTestBase {
     private final ClientLogger logger = new ClientLogger(ServiceBusReceiverAsyncClientIntegrationTest.class);
     private final AtomicInteger messagesPending = new AtomicInteger();
+    private final AtomicReference<List<Long>> messagesDeferredPending = new AtomicReference<>();
```

Comment:
Why not just an ArrayList? rather than an AtomicReference wrapping a list.

Reply:
Still unsure of why we are using an AtomicReference? Tests are run sequentially. Even then, why not a ConcurrentSet.